services/121-service/src/programs/programs.service.ts

```diff
@@ -401,22 +421,23 @@ export class ProgramService {
   }
 
   // This function takes a filled ProgramEntity and returns a filled ProgramReturnDto
-  private fillProgramReturnDto(program: ProgramEntity): ProgramReturnDto {
+  private fillProgramReturnDto(program: FoundProgram): ProgramReturnDto {
+    // XXX: what to do with all these nullable values? For now, forcing them to defined, but this is hacky
```

Comment:
IMO it makes more sense to fix the interface `FoundProgram` according to our data, But after checked it, I feel its more complicated 😄, may be for now its OK to keep it like this as hacky.

Reply:
I decided to change `ProgramReturnDto` instead, so that the fields that can be undefined are set to `IsOptional` given that this reflects "reality" better.